### DISCUSSION TITLE
extend `authenticator` methods

### DISCUSSION THREAD

**Discussion by rodbs** [sergiodxa/remix-auth#210] [Q&A] [answered]
`authenticator` exposes the `isAuthenticated` method.
I'd like to add more info to the user session (for instance, when a user is subscribed), and then to use a kind of 
`authenticator.isSubscribed` in the loader.
Is there an easy way to extend `authenticator` methods to add more checks? I'd appreciate it any inputs about it ! 
thx!

**Comment by sergiodxa** [comment, marked as answer]
You own the session storage, this means you can store anything in the session and grab it as usual in Remix

```ts
async function isSubscribed(request: Request) {
  let session = await sessionStorage.getSession(request.headers.get("cookie"))
  let isSubscribed = session.get("isSubscribed")
  if (typeof isSubscribed === "boolean") return isSubscribed
  return false;
}

export async function loader({ request }: LoaderArgs) {
  if (await isSubscribed(request)) {
    // do something
  } else {
    // do something else
  }
}
```

---

Another option is to store that data in the value returned by the strategy verify callback, then you can use the `isAuthenticated` method.

```ts
export async function loader({ request }: LoaderArgs) {
  let { isSubscribed } = await authenticator.isAuthenticated(request, {
    failureRedirect: "/login"
  })
  // your code here
}
```

**Reply by rodbs** [reply]
Ok, I've got it the first one.
But what do you mean by the 'strategy verify callback'?  I though `isAuthenticated` always returned the user object?

**Reply by sergiodxa** [reply]
Every strategy receives a function, that is the verify callback, there you return what your app considers the user object, you can return anything there, so you could add the isSubscribed boolean there.

**Reply by rodbs** [reply]
ok, understood. Thx!!
